src/pages/PassportRedemption/GiveawayPopover.tsx

```diff
@@ -0,0 +1,237 @@
+import React , { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import GreenCheck from './GreenCheck'
+import InstagramEnabled from './InstagramEnabled'
+import InstagramDisabled from './InstagramDisabled';
+
+import { getAllParticipatingSellers, getGiveawayTicketsForContact } from '../../utilities/api/interactionManager'
+
+
+interface Props {
+  showInstagram : boolean
+  contactId: string
+}
+
+interface TextProps {
+  icon ?: boolean
+}
+
+interface IgButtonProps {
+  disabled ?: boolean
+}
+interface GiveawayData {
+  weekly_giveaway_entries: String
+  number_of_visits_left : undefined | number
+}
+
+const GiveawayPopover = (props: Props) => {
+  const [showInfo, setShowInfo] = useState(false);
+  const [giveawayData, setData] = useState<GiveawayData>({
+    weekly_giveaway_entries: '',
+    number_of_visits_left: undefined
+  })
+  useEffect(() => {
+    async function setGiveawayCount(contactId: string) {
+      try {
+        const qtySellers = await getAllParticipatingSellers()
+          .then((res) => res.data.length);
+        const { weekly_giveaway_entries, unique_seller_tickets } = await getGiveawayTicketsForContact(contactId)
+          .then((res) => res.data);
+        setData({weekly_giveaway_entries, number_of_visits_left: qtySellers - unique_seller_tickets})
+      } catch (err) {
+        console.error(err);
+      }
+    }
+    if (props.showInstagram && showInfo) {
+      setGiveawayCount(props.contactId)
+    }
+  }, [props.showInstagram, showInfo, props.contactId])
+  return (
+    <PopoverContainer>
+      {props.showInstagram && showInfo && <PopoverContent>
+        <ToggleClosedButton onClick={() => setShowInfo(false)}>
+          <InstagramEnabled/>
+        </ToggleClosedButton>
+        <Title>
+          INSTAGRAM FOR GIVEAWAY ADDED
+        </Title>
+        <Divider/>
+        <GiveawayTextContainer>
+          <InfoText>
+            <TextBold>{giveawayData.weekly_giveaway_entries}</TextBold>
+            <GeneralText>
+              Weekly Giveaway Entries
+            </GeneralText>
+          </InfoText>
+
+          {giveawayData.number_of_visits_left !== 0 ?
+          <InfoText>
+            <TextBold>{giveawayData.number_of_visits_left}</TextBold>
+            <GeneralText>
+              Vendor Tickets away from a Grand Prize Giveaway Entry
```

Comment:
nit: maybe in your refactor, let's handle the translations of these

Reply:
Gotcha, who do I talk to about localization?

Reply:
@chelchan